src/GraphControl/Control/Grapher.cpp

```diff
@@ -118,13 +118,33 @@ namespace GraphControl
     {
         if (m_graph != nullptr && m_renderMain != nullptr)
         {
-            if (auto renderer = m_graph->GetRenderer())
+            if(auto renderer = m_graph->GetRenderer())
             {
-                if (SUCCEEDED(renderer->ResetRange()))
+                HRESULT hr;
+                if (m_resetUsingInitialDisplayRange)
+                {
+                    hr = renderer->SetDisplayRanges(m_initialDisplayRangeXMin, m_initialDisplayRangeXMax, m_initialDisplayRangeYMin, m_initialDisplayRangeYMax);
+                    m_resetUsingInitialDisplayRange = false;
+                }
+                else if (m_rangeUpdatedBySettings)
+                {
+                    IsKeepCurrentView = false;
+                    TryPlotGraph(false, false);
+                    m_rangeUpdatedBySettings = false;
+                    GraphViewChangedEvent(this, GraphViewChangedReason::Reset);
+                    return;
```

Comment:
This function is a little harder to follow, can we add a comment explaining when we would enter each block and noting that we return early in this one?